ci-operator/step-registry/rosa/nodes/wait-ready/rosa-nodes-wait-ready-commands.sh

```diff
@@ -10,24 +10,106 @@ log(){
+      oc describe node "$node"
+    done
+    log "$(date) - Finished printing details of all infra and worker nodes."
+}
+
+# Display details of machinesets and nodes for Classic Rosa cluster
+function listMachineAndNodeDetails() {
+    echo "List machine details"
+    # Get current machinesets and machines 
     log "$(date) - List of machinesets"
-    oc get machinesets -A
+    echo "oc get machinesets -A -n openshift-machine-api"
+    oc get machinesets -A -n openshift-machine-api
     log "$(date) - List of machines"
-    oc get machines -A | cat -n
-    log "$(date) - List of all worker only nodes, excluding infra"
-    oc get nodes --no-headers -l node-role.kubernetes.io/worker,node-role.kubernetes.io/infra!= | cat -n
+    echo "oc get machines -A -n openshift-machine-api | cat -n"
+    oc get machines -A -n openshift-machine-api | cat -n
 
-    # Get details of machines not in Running state
+    # Get details of worker machines not in Running state
+    log "$(date) - Worker machines not in Running state, if any"
     for machine in $(oc get machine  -n openshift-machine-api -l machine.openshift.io/cluster-api-machine-type=worker,machine.openshift.io/cluster-api-machine-type!=infra  --output jsonpath="{.items[?(@.status.phase!='Running')].metadata.name}"); do
       oc describe machine "$machine" -n openshift-machine-api
     done
-    # Get details of nodes not in Ready state
-    for node in $(oc get nodes --no-headers -l node-role.kubernetes.io/worker,node-role.kubernetes.io/infra!= --output jsonpath="{.items[?(@.status.conditions[-1].type!='Ready')].metadata.name}"); do
-      oc describe node "$node"
+    echo
+    
+    # Get details of all nodes 
+    listNodeDetails
+}
+
+# Display Machine Config Pool details
+function listMachineConfigPoolDetails() {
+    log "$(date) - List machine config pools"
+    echo "oc get mcp"
+    oc get mcp
+}
+
+# List details of machinesets, machines and nodes depending on Classic Rosa or HCP cluster
+function listDetails() {
+  listMachineConfigPoolDetails
+  if [ "$is_hcp_cluster" = "false" ]; then
+    echo "Listing machine and node details"
+    listMachineAndNodeDetails
+  else
+    echo "Listing node details only"
+    listNodeDetails
+  fi
+}
+
+# Function introducing sleep and wait for worker nodes to become Ready state
+function waitForReady() {
+    echo "Wait for all nodes to be Ready"
+
+    # Sleep for 2 minutes to allow for the nodes to begin restarting after initial scaling
+    sleep 120
+    # Query the node state until all of the nodes are ready
+    for i in {1..60}; do
+        NODE_STATE="$(oc get nodes || echo "ERROR")"
+        if [[ ${NODE_STATE} == *"NotReady"*  || ${NODE_STATE} == *"SchedulingDisabled"* ]]; then
+            FINAL_NODE_STATE="Fail"
+            echo "Not all of the nodes have finished restarting - waiting for 30 seconds, attempt ${i}"
+            sleep 30
+        elif [[ ${NODE_STATE} == "ERROR" ]]; then
```

Comment:
If `oc get nodes` meets error, it is meaningless to run 30m, let us terminate the command are retrying 3 times.  And in order to improve the readability, let us put it into an individual block. The code structure should be like this:
`if ERROR; then 
    retry++
    sleep 30
    continue
  fi  
  If retry > 3; then
    #do  something
    break
  fi
     `